### PR TITLE
Added flask routes and html templates

### LINKED ISSUE #14: Create Flask routes for the dashboard
1. Routes for the dashboard landing page
2. Routes to dynamically update the progress bars

### LINKED ISSUE #15: Create Flask routes for submitting new scans

### LINKED ISSUE #17: Setup MongoDB server and connect the flask app to it

### LINKED ISSUE #18: Design a schema for each entry in the database

### PR DESCRIPTION
**Pull request by maanas-talwar**
Closes #14, closes #15, closes #17, closes #18

@kmehant @Ammoniya @1209973 Please review these changes and share your feedback for the same.
Thanks

### CODE REVIEW

**Review by kmehant — CHANGES_REQUESTED** [review]
@maanas-talwar 

1. Can you post screenshots for each endpoint you have defined?
2. Can you refactor the present approach using `add_url_rule()` (why? decouples annotations and view functions much better for maintaining source code in the future)
3. Please explicitly mention the HTTP methods while configuring the routes
4. Add a TODO: tag for adding caching layers to routes in the future (at the end of the project if time permits or maybe in the future)
5. Isn't this part of the Master node, or are we having a separate server component for Master?

File: Dashboard/templates/index.html [hunk 1]
```diff
@@ -21,140 +21,77 @@
...
                     </div>
                 </div>
             </div>
+            {% endfor %}
         </div>
     </div>
+
+    <script>
+        function openUploader() {
+            window.location = "http://localhost:5000/newScan";
```
Comment (1209973) [inline Dashboard/templates/index.html:92]: @maanas-talwar if we can get the base URL from the env variables it would be better.

**Review by 1209973 — CHANGES_REQUESTED** [review]

**Comment by Ammoniya** [comment]
You have to create separate endpoint for the progress bar, else the status of the progress bar will be static or have to reload the frontpage constantly.

Ref - https://github.com/djdmorrison/flask-progress-example

**Comment by maanas-talwar** [comment]
@kmehant I believe the part to service requests should be separate from the coordinator node.
The coordinator node will be responsible only for defining the jobs and adding them to the queue.
As for the other stuff, I will make the required changes and ping you guys if I'm stuck anywhere.
Thanks

File: Dashboard/templates/index.html (hunk 1 above)
Comment (maanas-talwar) [reply Dashboard/templates/index.html:92]: Sure

**Comment by maanas-talwar** [comment]
@Ammoniya I will surely try to include this.

**Comment by maanas-talwar** [comment]
@Ammoniya I tried to incorporate the code that you suggested for a dynamic progress bar.
I believe we will have to create a /progress/<id> route as then only each progress bar will receive a stream of data independent of others. Does this seem fine?

**Comment by kmehant** [comment]
> @kmehant I believe the part to service requests should be separate from the coordinator node.
> The coordinator node will be responsible only for defining the jobs and adding them to the queue.
> As for the other stuff, I will make the required changes and ping you guys if I'm stuck anywhere.
> Thanks

@maanas-talwar Great! 

Also some useful links for you regarding progress bar 😄 
* https://stackoverflow.com/questions/24251898/flask-app-update-progress-bar-while-function-runs
* https://medium.com/@sureshd_731/how-to-generate-multiple-progress-bars-in-a-flask-app-8e77c013a81d

**Comment by maanas-talwar** [comment]
@kmehant Thanks for this. The medium article actually uses the same code that @Ammoniya mentioned XD
I will research more on this and get back to you with the best approach as updating the progress every time the values are changed in the database seems better than attaching listeners between frontend and flask app & flask app and database.

**Comment by maanas-talwar** [comment]
@Ammoniya @kmehant @1209973  I have implemented the dynamic progress bar to some extent. I will set up and connect the MongoDB instance to the app to test it.

**Comment by kmehant** [comment]
@maanas-talwar Great Work! I truly appreciate your efforts. Looking forward to upcoming updates!

**Comment by kmehant** [comment]
@maanas-talwar can you add screenshots?

**Comment by maanas-talwar** [comment]
> @maanas-talwar can you add screenshots?

Sure

**Comment by maanas-talwar** [comment]
Dashboard route:
dashboard
New Scan route:
newScan

**Comment by kmehant** [comment]
@Ammoniya @1209973 I have few refactoring suggestions on this PR. However, I feel we can achieve that even at the end. Shall we approve this as a good initial code?

**Review by 1209973 — APPROVED** [review]

**Review by kmehant — APPROVED** [review]

**Review by Ammoniya — APPROVED** [review]

**Pull request merged**